spec/tasks/init_spec.rb

```diff
@@ -143,4 +143,64 @@
       expect(stub).to have_been_requested.once
     end
   end
+
+  context 'when an empty response body is received' do
+    before(:each) do
+      stub_request(:post, 'http://0.0.0.0/post')
+        .to_return(status: response_status, body: nil)
+    end
+
+    let(:opts) {
+      {
+        method:   'post',
+        base_url: "#{url}/post"
+      }
+    }
+
+    [101, 204, 304].each do |iter_status|
```

Comment:
If we just handle `nil` response bodies instead of specific response codes, we can just do a single test instead of iterating over response codes.